plugins/inputs/http_response/http_response.go

```diff
@@ -361,6 +366,14 @@ func (h *HTTPResponse) httpGather(u string) (map[string]interface{}, map[string]
 			setResult("response_string_mismatch", fields, tags)
 			fields["response_string_match"] = 0
 		}
+	} else if h.ResponseStatusCodeMatch > 0 {
```

Comment:
I don't think you want an else here.  That would make the response code check only happen if "response_string_match" is empty.

If you remove the else here, give the block its own comment and undo the comment change above.
```suggestion
	}
	// Check the reponse status code
	if h.ResponseStatusCodeMatch > 0 {
```

Reply:
I've done the change, my little worry here is when the both condition fail, the `result` tag will have `response_status_code_mismatch` which it's can be confusing because isn't just this problem